### PR TITLE
feat(GuildMember): add 'resetNickname' method

### PR DESCRIPTION
**Pull request by jecsham**
**Please describe the changes this PR makes and why it should be merged:**

We know that there are alternatives to reset a nicknames, like:
```js
<GuildMember>.setNickname('');
// or
<GuildMember>.setNickname(<GuildMember>.user.username) // not recommended
```
however, not many of us know this and it may take some time for some to realize it, as has been my case. That's why I decided to create this alias.

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by anandre** [comment]
I would argue that better documentation on the `setNickname` method would be preferable instead of making a helper method to do it.  Passing `null` to the method will reset it.

tl;dr: make a PR to make documentation/example better and show using `null` not a whole new helper method

**Comment by jecsham** [comment]
@anandre I had thought about it, but I think its more intuitive with a helper function. 
With `setNickname` it would still be confusing to figure out. I'm not saying that consulting the documentation is bad, but usually many people uses typescript, so they can see methods and properties that an object has, which saves time.

I understand that the helper functions are usually redundant code, but they make a library much easier to use.

Anyway, if you don't change your mind, @zaida04 started the PR you requested, so you can close this one without problems!

**Comment by iCrawl** [comment]
See #5070
